packages/backend/src/modules/bridges/BridgeComparator.ts

```diff
@@ -0,0 +1,113 @@
+import type { Logger } from '@l2beat/backend-tools'
+import type { Database } from '@l2beat/database'
+import type { BridgeComparePlugin } from './compare/types'
+
+interface BridgeCompareConfig {
+  intervalMs?: number
+  timeoutMs?: number
+}
+
+export class BridgeComparator {
+  private running = false
+  private readonly config: Required<BridgeCompareConfig>
+
+  constructor(
+    private db: Database,
+    private plugins: BridgeComparePlugin[],
+    private logger: Logger,
+    options: BridgeCompareConfig = {},
+  ) {
+    this.logger = logger.for(this)
+    this.config = {
+      intervalMs: options.intervalMs ?? 10 * 60_000,
+      timeoutMs: options.timeoutMs ?? 10_000,
+    }
+  }
+
+  start() {
+    const runCompare = async () => {
+      if (this.running) {
+        return
+      }
+      this.running = true
+      try {
+        await this.compare()
+      } catch (e) {
+        this.logger.error(e)
+      }
+      this.running = false
+    }
+    setInterval(runCompare, this.config.intervalMs)
+    this.logger.info('Compare schedulded', {
+      intervalMs: this.config.intervalMs,
+    })
+    runCompare()
+    this.logger.info('Started')
+  }
+
+  async compare() {
+    this.logger.info('Running compare', {
+      plugins: this.plugins.map((p) => p.name),
+    })
+    this.logger.info('Fetching items from external explorers...', {
+      plugins: this.plugins.map((p) => p.name),
+    })
+    const items = (
+      await Promise.all(
+        this.plugins.map(async (plugin) => {
+          try {
+            return {
+              plugin,
+              items: await plugin.getExternalItems(),
+            }
+          } catch (error) {
+            this.logger.warn(`Plugin for ${plugin.name} failed`, error)
+            return null
+          }
+        }),
+      )
+    ).filter((x) => x != null)
+
+    this.logger.info('Setting timeout...', { timeout: this.config.timeoutMs })
+    // This timeout is needed to make sure our backend indexes latest events
+    await new Promise((resolve) => setTimeout(resolve, this.config.timeoutMs))
```

Comment:
Again, I'd have a separate time loop. So one loop for getting the hash pairs and another for checking.
When you find a pair you remember the hashes but also remember the time.
When you check the pairs and you find matching ones you just remove them from the list of remembered.
After checking if there are leftover found pairs with age > 5/10 mins then alert